### PR TITLE
#26020 Fixed type annotations for pandas.plotting._core

### PR DESCRIPTION
**Pull request by gwrome**
- [X] closes ##26020
- [X] tests passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -1723,7 +1724,8 @@ def result(self):
 _all_kinds = _common_kinds + _dataframe_kinds + _series_kinds
 
 _klasses = [LinePlot, BarPlot, BarhPlot, KdePlot, HistPlot, BoxPlot,
-            ScatterPlot, HexBinPlot, AreaPlot, PiePlot]
+            ScatterPlot, HexBinPlot, AreaPlot, PiePlot] \
+    # type: List[Type[MPLPlot]]
```
Comment (WillAyd) [inline pandas/plotting/_core.py]: Should just be `List[MPLPlot]` I think - does that give you some kind of error?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/plotting/_core.py (hunk 1 above)
Comment (gwrome) [reply pandas/plotting/_core.py]: Yes, `_klasses` contains the classes/types themselves, not instances of those types, so using `List[MPLPlot]` errors out with lines like `error: List item 0 has incompatible type "Type[LinePlot]"; expected "MPLPlot"`.

**Comment by jreback** [comment]
lgtm. @WillAyd

File: mypy.ini [hunk 2]
```diff
@@ -167,7 +167,17 @@ ignore_errors=True
...
+[mypy-pandas.tseries.frequencies]
+ignore_errors=True
+
+[mypy-pandas.tseries.holiday]
+ignore_errors=True
+
+[mypy-pandas.tseries.offsets]
+=======
 [mypy-pandas.plotting._core]
+>>>>>>> origin/master
```
Comment (gfyoung) [inline mypy.ini]: Merge conflict vestiges?

Comment (gwrome) [reply mypy.ini]: Oops. Changed IDEs, forgot to turn on autosave.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @gwrome keep them coming!

**Pull request merged**